pxr/usd/usd/stageCache.h

```diff
@@ -107,15 +106,23 @@ class UsdStageCache
         /// Create an Id from a string value.  The supplied \p val must have
         /// been obtained by calling ToString() previously.
         static Id FromString(const std::string &s) {
-            return FromLongInt(boost::lexical_cast<long int>(s));
+            bool overflow = false;
+            int64_t result = TfStringToInt64(s, &overflow);
```

Comment:
Using `long result = TfStringToLong(s, &overflow);` would be more technically correct here.